app/src/main/java/com/infomaniak/mail/utils/extensions/Extensions.kt

```diff
@@ -416,7 +418,22 @@ fun List<Folder>.addDividerBeforeFirstCustomFolder(dividerType: Any): List<Any>
 //endregion
 
 //region Messages
-fun List<Message>.getFoldersIds(exception: String? = null) = mapNotNull { if (it.folderId == exception) null else it.folderId }
+fun List<Message>.getFoldersIds(exception: String? = null): ImpactedFolders {
+    val impactedFolders = ImpactedFolders()
+
+    forEach { message ->
+        when {
+            message.folderId == exception -> Unit
+            message.snoozeState == SnoozeState.Snoozed -> {
+                impactedFolders += message.folderId
+                impactedFolders += FolderRole.SNOOZED
+            }
+            else -> impactedFolders += message.folderId
+        }
+    }
```

Comment:
```suggestion
    for (message in this) {
        if (message.folderId == exception) continue
        impactedFolders += message.folderId
        if (message.snoozeState == SnoozeState.Snoozed) impactedFolders += FolderRole.SNOOZED
    }
```

Reply:
`impactedFolders += message.folderId` was duplicated, and I felt it was kinda hidden that it was supposed to be done every time.

Reply:
Yeah I agree, it's nice to show that it needs to be done *every* time with the way you wrote it